pkg/util/watch_filters.go

```diff
@@ -79,6 +81,38 @@ func ResolveCAPIMachineSetFromInfraMachineTemplate(namespace string) func(contex
 	}
 }
 
+// ResolveCAPIMachineFromInfraMachine resolves a CAPI Machine from an InfraMachine. It takes client.Object,
+// and uses owner references to determine the owning CAPI machine. If one is found, it returns a reconcile.Request
+// for the corresponding MAPI Machine in the MAPI namespace to trigger reconciliation of the mirror MAPI Machine.
+func ResolveCAPIMachineFromInfraMachine(namespace string) func(context.Context, client.Object) []reconcile.Request {
+	return func(ctx context.Context, obj client.Object) []reconcile.Request {
+		logger := logf.FromContext(ctx).WithValues(
+			"objectType", fmt.Sprintf("%T", obj),
+			"namespace", obj.GetNamespace(),
+			"name", obj.GetName(),
+		)
+		logger.V(4).Info("reconcile triggered")
```

Comment:
Wouldn't we normally log this in the controller instead of here?

Reply:
yea, this is just a log line to output the values we set above